bundles/plugins/org.bonitasoft.studio.importer.bar/src/org/bonitasoft/studio/importer/bar/custom/migration/WidgetMigration.java

```diff
@@ -106,25 +113,41 @@ public void migrateBefore(final Model model, final Metamodel metamodel)
                 storeDisplayAfterEventDependsOnConditionScripts(widget);
                 storeHelpMessages(widget);
                 storeInjectWidgetScripts(widget);
-                storeInputConnectors(widget);
-                storeAfterEventConnectors(widget);
+                storeInputConnectors(model, widget);
+                storeAfterEventConnectors(model, widget);
             }
         }
     }
 
-    private void storeInputConnectors(final Instance widget) {
+    private void storeInputConnectors(final Model model, final Instance widget) {
         final List<Instance> connectors = widget.get("inputConnectors");
         if (!connectors.isEmpty()) {
             final Instance instance = connectors.get(0);
             initialValueConnectors.put(widget.getUuid(), instance.copy());
+            if (connectors.size() > 1) {
```

Comment:
I think I would have extract a method removeExtraConnectors or something in order to help understanding why we are doing it